sdk/appcenter/src/main/java/com/microsoft/appcenter/AppCenter.java

```diff
@@ -1170,6 +1182,24 @@ public void run() {
         return future;
     }
 
+    /**
+     * Implements {@link #setAuthToken} at instance level.
+     */
+    private void setInstanceAuthToken(String authToken) {
+        AuthTokenContext authTokenContext = AuthTokenContext.getInstance();
+        JwtClaims claims = null;
+        if (authToken != null) {
+            claims = JwtClaims.parse(authToken);
+        }
+        if (claims != null) {
```

Comment:
You could invert the condition.

Reply:
I prefer this organization, but I suppose it's a stylistic choice

Reply:
We can get rid of unnecessary one bit operation there. 😄

Reply:
Not really the logging matters here.
If you pass null we just log that you remove the token, it's not an error.
If you pass an invalid JWT, we log an error and utils AND we remove the token.

Reply:
It's more common to have the success block before the failure one in a if/else. Both work.